components/pages/edit/editor/Editor.tsx

```diff
@@ -0,0 +1,83 @@
+import { useRef, useEffect, useState } from 'react';
+import { Editor } from '@toast-ui/react-editor';
+import { imageUpload } from '../../../../hooks/ImageUpload';
+import { handleDrop } from '../../../../hooks/FileDrop';
+import '@nance/nance-editor/lib/editor.css';
+
+function TextEditor({
+  onChange,
+  initialValue,
+}: {
+  onChange: (value: string) => void,
+  initialValue?: string,
+}) {
+  const editorRef = useRef<Editor>(null);
+
+  const [imageUploading, setImageUploading] = useState(0);
+
+  const simulateLoading = (setLoading: (progress: number) => void, fileSize: number) => {
+    const time = Math.round(fileSize / (10 * 1000)) * 10;
+    let progress = 0;
+    const interval = setInterval(() => {
+      progress += 10;
+      if (progress > 100) {
+        clearInterval(interval);
+      } else {
+        setLoading(progress);
+      }
+    }, time);
+    return interval;
+  };
+
+  useEffect(() => {
+    const editor = editorRef.current?.getInstance();
+    const initDropHandler = async () => {
+      const md = await handleDrop();
+      if (editor && md) editor.setMarkdown(md);
```

Comment:
this looks strange to me: the effect happen after render, this handleDrop set event listener and why would editor need to wait it returns md?
 
 Can we add handleFileDrop with `onDrop` field of Editor component? something like:
 ```
 <Editor 
  ...
  onDrop={(e) => {
   const ipfsUri = upload(e.file);
   setMarkdown(...)
 }} />
 ```
 this way we can reduce the usage of useEffect which can bring bugs.